### PR TITLE
ISS refactor - batch 3

### PR DESCRIPTION
**Pull request by vsedmik**
Based on #12347.

Introduced changes:
1. Extended `test_positive_import_content_for_disconnected_sat_with_existing_content` by assertion for correct error message when non-import-only CV is used for import
2. Removed `_import_entities`
3. Added test for customer scenarios:
   ~`test_postive_export_import_cv_with_long_name`~
   `test_positive_export_rerun_failed_import`
   `test_postive_export_import_repo_with_GPG`

### CODE REVIEW

**Comment by vsedmik** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_satellitesync.py -k 'test_positive_import_content_for_disconnected_sat_with_existing_content or test_postive_export_import_cv_with_long_name or test_positive_export_rerun_failed_import or test_postive_export_import_repo_with_GPG'

**Comment by vsedmik** [comment]
rebased

**Comment by vsedmik** [comment]
rebased on Jake's ruff

**Comment by vsedmik** [comment]
rebased on Jake's ruff and my other PR

**Comment by vsedmik** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_satellitesync.py -k 'test_positive_import_content_for_disconnected_sat_with_existing_content or test_postive_export_import_cv_with_long_name or test_positive_export_rerun_failed_import or test_postive_export_import_repo_with_GPG'

File: tests/foreman/cli/test_satellitesync.py [hunk 1]
```diff
@@ -1333,6 +1301,184 @@ def test_postive_export_import_cv_with_file_content(
...
+        :steps:
+            1. Create CV, add repo from the setup, publish it and run export.
+            2. Start import of the CV into another organization and kill it before it's done.
+            3. Rerun the import again, let it finish and check the CVV was imported.
+
+        :expectedresults:
+            1. First import should fail, no CVV should be added.
+            2. Second import should succeed without errors and should contain the CVV.
+
+        :CaseImportance: Medium
```
Comment (jyejare) [inline tests/foreman/cli/test_satellitesync.py:1414]: This looks like a high-importance case as the possibilities are high for such case in customer env !

File: tests/foreman/cli/test_satellitesync.py [hunk 2]
```diff
@@ -1480,13 +1693,15 @@ def test_positive_import_content_for_disconnected_sat_with_existing_content(
...
         :steps:
-            1. Run complete export of the CV.
-            2. On Disconnected satellite, create a cv with same name as cv on 2 and with
-               'import-only' selected.
-            3. Run the import command.
+            1. Run complete export of the CV from setup.
+            2. On Disconnected satellite, create a CV with the same name as setup CV and with
+               'import-only' set to False and run the import command.
+            3. On Disconnected satellite, create a CV with the same name as setup CV and with
+               'import-only' set to True and run the import command.
```
Comment (jyejare) [inline tests/foreman/cli/test_satellitesync.py:1700]: Shouldnt this be two different cases or parameterized ? Becasue failing step 2 will skip skip step 3 assertion.

**Review by jyejare — COMMENTED** [review]
ACK pending comments/questions!

File: tests/foreman/cli/test_satellitesync.py [hunk 3]
```diff
@@ -1333,6 +1301,184 @@ def test_postive_export_import_cv_with_file_content(
         assert len(imported_files)
         assert len(exported_files) == len(imported_files)
 
+    @pytest.mark.tier3
+    def test_postive_export_import_cv_with_long_name(
```
[the remark this answers is not in the document]
Comment (vsedmik) [reply tests/foreman/cli/test_satellitesync.py:1305]: This is a great question actually - what are the criteria? Are there any common or general? I haven't heard of any.

Basically BZs proposed by QE/SD/DEV were brought in during the eval, added as "new" cases to be automated, sprint was planned and that was it. So I considered it rather "TODO" insted of "nope" :/

File: tests/foreman/cli/test_satellitesync.py (hunk 1 above)
Comment (vsedmik) [reply tests/foreman/cli/test_satellitesync.py:1414]: Here I just followed the severity reported at that BZ, which is medium.

File: tests/foreman/cli/test_satellitesync.py (hunk 2 above)
Comment (vsedmik) [reply tests/foreman/cli/test_satellitesync.py:1700]: Well, it could. On the other hand here we leverage a single export to check the import behaves as expected in both cases, so we follow the "do more with less" strategy. To me it feels like... accurate as is.

File: tests/foreman/cli/test_satellitesync.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (vijaysawant) [reply tests/foreman/cli/test_satellitesync.py:1305]: @jyejare @vsedmik we are considering BZ's which has high, urgent sevirity with CC attached to it.
There are few bz's around `content-view export/import` command failing for larger name (in CV, product, repo) with different scenarios like docker imange pull, repo sync, etc.
I guess good to add such tests in negative scenarios rather in positive.

**Review by sambible — APPROVED** [review]

**Review by jyejare — APPROVED** [review]

**Review by damoore044 — APPROVED** [review]
Looks good, merging

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_satellitesync.py -k 'test_positive_import_content_for_disconnected_sat_with_existing_content or test_postive_export_import_cv_with_long_name or test_positive_export_rerun_failed_import or test_postive_export_import_repo_with_GPG'
